RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -1004,149 +953,148 @@ void MuonIdProducer::fillArbitrationInfo( reco::MuonCollection* pOutputMuons )
+                      fabs(segment2.dXdZErr - segment1.dXdZErr) < 1E-3 &&
+                      fabs(segment2.dYdZErr - segment1.dYdZErr) < 1E-3)
+                   {
+                     arbitrationPairs.push_back(std::make_pair(&chamber2, &segment2));
+                   }
+                 } // segmentIter2
+               } // chamberIter2
+             } // muonIndex2
+           }
+
+           // arbitration segment sort
+           if(arbitrationPairs.empty()) continue; // this should never happen
+           if(arbitrationPairs.size()==1) {
+             arbitrationPairs.front().second->setMask(reco::MuonSegmentMatch::BelongsToTrackByDRSlope);
+             arbitrationPairs.front().second->setMask(reco::MuonSegmentMatch::BelongsToTrackByDXSlope);
+             arbitrationPairs.front().second->setMask(reco::MuonSegmentMatch::BelongsToTrackByDR);
+             arbitrationPairs.front().second->setMask(reco::MuonSegmentMatch::BelongsToTrackByDX);
+             arbitrationPairs.front().second->setMask(reco::MuonSegmentMatch::Arbitrated);
+           } else {
+             sort(arbitrationPairs.begin(), arbitrationPairs.end(), SortMuonSegmentMatches(reco::MuonSegmentMatch::BelongsToTrackByDRSlope));
+             arbitrationPairs.front().second->setMask(reco::MuonSegmentMatch::BelongsToTrackByDRSlope);
+             sort(arbitrationPairs.begin(), arbitrationPairs.end(), SortMuonSegmentMatches(reco::MuonSegmentMatch::BelongsToTrackByDXSlope));
+             arbitrationPairs.front().second->setMask(reco::MuonSegmentMatch::BelongsToTrackByDXSlope);
+             sort(arbitrationPairs.begin(), arbitrationPairs.end(), SortMuonSegmentMatches(reco::MuonSegmentMatch::BelongsToTrackByDR));
+             arbitrationPairs.front().second->setMask(reco::MuonSegmentMatch::BelongsToTrackByDR);
+             sort(arbitrationPairs.begin(), arbitrationPairs.end(), SortMuonSegmentMatches(reco::MuonSegmentMatch::BelongsToTrackByDX));
+             arbitrationPairs.front().second->setMask(reco::MuonSegmentMatch::BelongsToTrackByDX);
+             for ( auto& ap : arbitrationPairs ) {
+               ap.second->setMask(reco::MuonSegmentMatch::Arbitrated);
+             }
+           }
          }
-      } // chamberIter1
 
-      // station segment sort
-      for( int stationIndex = 1; stationIndex < 5; ++stationIndex )
-         for( int detectorIndex = 1; detectorIndex < 4; ++detectorIndex )
+         // setup me1a cleaning for later
+         if( chamber1.id.subdetId() == MuonSubdetId::CSC && arbClean_ &&
+             CSCDetId(chamber1.id).ring() == 4) {
+           for ( auto& segment2 : chamber1.segmentMatches ) {
+             if( segment1.cscSegmentRef.isNull() || segment2.cscSegmentRef.isNull() ) continue;
+             if( meshAlgo_->isDuplicateOf(segment1.cscSegmentRef,segment2.cscSegmentRef) &&
+                 (segment2.mask & 0x1e0000) &&
+                 (segment1.mask & 0x1e0000) ) {
```

Comment:
A constant with a meaningful name would be better than this magic number 0x1e0000.